src/common/device/AppleDevice.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { Logger } from '@salesforce/core';
+import { Version } from '../Common.js';
+import { IOSUtils } from '../IOSUtils.js';
+import { BaseDevice, DeviceType, LaunchArgument } from './BaseDevice.js';
+
+export type AppleRuntime = {
+    bundlePath: string;
+    identifier: string;
+    isAvailable: boolean;
+    isInternal: boolean;
+    name: string;
+    platform: string;
+    runtimeRoot: string;
+    version: string;
+};
+
+export enum AppleOSType {
+    iOS = 'iOS',
+    tvOS = 'tvOS',
+    watchOS = 'watchOS',
+    xrOS = 'xrOS'
+}
+
+export class AppleDevice implements BaseDevice {
+    public logger?: Logger;
+    public readonly id: string;
+    public readonly name: string;
+    public readonly deviceType: DeviceType;
+    public readonly osType: string;
+    public readonly osVersion: string | Version;
+
+    public constructor(
+        id: string,
+        name: string,
+        deviceType: DeviceType,
+        osType: string,
+        osVersion: Version,
+        logger?: Logger
+    ) {
+        this.id = id;
+        this.name = name;
+        this.deviceType = deviceType;
+        this.osType = osType;
+        this.osVersion = osVersion;
+        this.logger = logger;
+    }
+
+    /**
+     * A string representation of an AppleDevice which includes Device Name, OS Type, OS Version, and udid
+     */
+    public toString(): string {
+        return `${this.name}, ${this.osType} ${this.osVersion.toString()}, ${this.id}`;
+    }
+
+    /**
+     * Attempts to boot up the device.
+     *
+     * @param waitForBoot Optional boolean indicating whether to wait for the device to boot up. Defaults to true.
+     */
+    public async boot(waitForBoot = true): Promise<void> {
+        // Booting the device will not give it a window. We also need to launch
+        // the simulator app in order for the booted device(s) to get a window.
+        await IOSUtils.bootDevice(this.id, waitForBoot, this.logger);
+        await IOSUtils.launchSimulatorApp(this.logger);
+    }
+
+    /**
+     * Attempts to reboot the device.
+     *
+     * @param waitForBoot Optional boolean indicating whether to wait for the device to boot up. Defaults to true.
+     */
+    public async reboot(waitForBoot = true): Promise<void> {
+        // `xcrun simctl` does not provide a way to reboot a device.
+        // Instead we need to shutdown and again start up a device.
+        await this.shutdown();
+        await this.boot(waitForBoot);
+    }
+
+    /**
+     * Attempts to shutdown the device.
+     */
+    public async shutdown(): Promise<void> {
+        await IOSUtils.shutdownDevice(this.id, this.logger);
```

Comment:
Do you think there's value to keeping these underlying methods in the utils classes? Or would it make more sense to just inline their implementations in this class? To me I guess it would depend on whether there's value to calling those methods _outside of_ this class.

Reply:
That's exactly what I was asking myself the other day :) I think I would like to move them out of the util classes in the near future. There are still other operations (like createDevice) that lives in the util classes and so when I get to moving that code in a follow up PR, I'll be moving these too (along with test coverage for these)